library/std/src/os/windows/io/socket.rs

```diff
@@ -71,8 +71,8 @@ impl BorrowedSocket<'_> {
     /// `INVALID_SOCKET`.
     #[inline]
     #[unstable(feature = "io_safety", issue = "87074")]
-    pub unsafe fn borrow_raw(socket: RawSocket) -> Self {
-        debug_assert_ne!(socket, c::INVALID_SOCKET as RawSocket);
+    pub const unsafe fn borrow_raw(socket: RawSocket) -> Self {
+        assert!(socket != c::INVALID_SOCKET as RawSocket);
```

Comment:
was `debug_assert_ne`, but became `assert`, so will be called at not debug builds. Why this change?

Reply:
This makes it consistent with the unix version of `borrowed_fd`, which does a plain `assert`. Which is there because the previous unix implementation code `FileDesc::new` used a plain `assert`.

Also, I think it makes sense; this is a new API, and we're expecting to migrate mature codebases to use it, so it makes sense to be extra careful that old and new code agree on when and where `INVALID_SOCKET` can appear. Also, `OwnedSocket` uses `rustc_layout_scalar_valid_range_*` layout tricks that assume that `INVALID_SOCKET` can't appear, so it could easily produce some dramatic UB if the invariant is violated.

Reply:
Make sense.